journals/apps/journals/views.py

```diff
@@ -84,3 +97,106 @@ def get(self, request, *args, **kwargs):
+        context = {
+            'url': reverse('admin:import_users', args=[journal_id]),
+            'journal_access_form': journal_form if journal_form else UsersJournalAccessForm(
+                initial={'expiration_date': get_default_expiration_date(journal)}
+            ),
+            'opts': journal._meta,
+        }
+        context.update(admin.site.each_context(request))
+        return context
+
+    def get_or_create_journal_users(self, request, usernames):
+        """
+        Checks whether the usernames are valid (do exist in journal database)
+        if a username doesn't exist then fetch its data from LMS and create
+        a user in journal.apps.models.USER
+
+        Returns:
+            returns the set of valid usernames (exist in journal database)
+        """
+        all_users = User.objects.all()
+        non_existing_users = [username for username in usernames if not all_users.filter(username=username).exists()]
+        valid_usernames = usernames - set(non_existing_users)
+        start, offset = 0, settings.BATCH_SIZE_FOR_LMS_USER_API
+
+        while start < len(non_existing_users):
+            users = set()
+            users_lms_data = User.account_details(
+                request,
+                ",".join(non_existing_users[start:start + offset])
+            )
+            for user_data in users_lms_data:
+                username = user_data['username']
+                valid_usernames.add(username)
+                users.add(User(
+                    username=username,
+                    email=user_data.get('email', ""),
+                    is_active=user_data.get('is_active', "False")
+                ))
+                log.info("'{}' user has been added in journal.".format(username))
+            User.objects.bulk_create(list(users))
+            start = start + offset
+        return valid_usernames
+
+    def handle_users_import(self, request, journal_id, journal_access_form):
+        """
+        Give journal's access to given users.
+        """
+
+        usernames = set(
+            parse_csv(
+                journal_access_form.cleaned_data['bulk_upload_csv']
+            )
+        )
+        usernames.remove('usernames')
+        valid_usernames = self.get_or_create_journal_users(request, usernames)
+        JournalAccess.bulk_create_journal_access(
+            valid_usernames,
+            journal_id,
+            journal_access_form.cleaned_data['expiration_date']
+        )
+
+        skipped_usernames = usernames - valid_usernames
+        if skipped_usernames:
+            log.warning("Following usernames have been skipped: '{}'".format(", ".join(list(skipped_usernames))))
+        return valid_usernames, skipped_usernames
+
+    def get(self, request, journal_id):
+        """
+        GET request method for JournalAccessView.
+        """
+        return render(request, self.template, self.get_context(request, journal_id))
+
+    def post(self, request, journal_id):
```

Comment:
I'm fine with either handling incorrect `journal_id`s further in where I mentioned them, but it may be easier to just check  at the beginning of the `post` or `get` methods so we can exit early.

Reply:
I am going with the start checks. It will make things more readable.